### PR TITLE
Implementation of basic coordinator

### PR DESCRIPTION
**Pull request by connormuench**
Addresses #1.

### CODE REVIEW

File: coordinator/coordinator.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    for rule in rules:
+        failed = False
+        subbed_expression = rule.expression
+        for match in re.finditer('{([A-Fa-f0-9]+)}', rule.expression):
+            # Replace all occurrences of point addresses (enclosed in braces) in the expression with their most recent value
+            if match.group(0) not in last_reading:
+                # At least one of the addresses specified in the expression doesn't have a last reading, so the expression can't be evaluated
+                failed = True
+                break
+            subbed_expression = re.sub('{([A-Fa-f0-9]+)}', last_reading[match.group(0)], subbed_expression, count=1)
```
Comment (wisp4400) [inline coordinator/coordinator.py:69]: not sure what this line does exactly.does it replaces the pointers in the rule with last data?

File: coordinator/dbinit.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+CONFIG_PATH = 'coordinator.json'
+
+# Load up the config file to get the db path
+with open(CONFIG_PATH) as config_file:
+    env = json.load(config_file)
+    
+db_filename = path.abspath(env['local-db-path'])
+
+if path.exists(db_filename):
+    os.remove(db_filename)
```
Comment (wisp4400) [inline coordinator/dbinit.py:13]: this is removing old versions of the DB?

**Review by wisp4400 — APPROVED** [review]

File: coordinator/coordinator.py (hunk 1 above)
Comment (connormuench) [reply coordinator/coordinator.py:69]: Yes, it uses regex to sub all addresses contained within braces with the current value of the device at that address.

File: coordinator/dbinit.py (hunk 2 above)
Comment (connormuench) [reply coordinator/dbinit.py:13]: Yes. Initialization of the database means that the old one is removed.

**Pull request merged**